examples/compiler/01_iet.ipynb

```diff
@@ -0,0 +1,664 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tutorial describes one particular intermediate representation used by the Devito Compiler: the Iteration/Expression Tree (IET), a special type of Abstract Syntax Tree (AST).\n",
+    "\n",
+    "# Part I - Top Down\n",
+    "\n",
+    "First, let's describe a _domain_, and a _function_ that will allow us to specify how such a domain gets modified. In particular, we will look at Functions that change through _time_. \n",
+    "\n",
+    "Thus we need a `Function` object with which we can build a timestepping scheme. For this purpose Devito provides so-called `TimeFunction` objects that encapsulate Functions that are differentiable in space and time, which are derived from basic _SymPy_ Functions. \n",
+    "\n",
+    "With this we can derive symbolic expressions for the backward derivatives in space directly via the `u.dxl` and `u.dyl` shorthand expressions (the l indicates \"left\" or backward differences) and the shorthand notation `u.dt` provided by `TimeFunction` objects to derive the forward derivative in time.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Data([[[0., 0., 0.],\n",
+       "       [0., 0., 0.],\n",
+       "       [0., 0., 0.]],\n",
+       "\n",
+       "      [[0., 0., 0.],\n",
+       "       [0., 0., 0.],\n",
+       "       [0., 0., 0.]]], dtype=float32)"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from devito import Eq, Grid, TimeFunction, Operator\n",
+    "\n",
+    "grid = Grid(shape=(3, 3))\n",
+    "u = TimeFunction(name='u', grid=grid)\n",
+    "u.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, we have just declared a two-dimensional domain with three coordinates at each dimension (_x_ and _y_). Each coordinate of such a discrete space will be holding a real value.\n",
+    "\n",
+    "As we can see, we can always access the values at each coordinate of the domain. At this point, no modifications have been done to it yet. `u.data` give us a quick access over the values holded by each cell of such a domain. `u.data[0]` holds the values in the grid at the \"current\" iteration time, given a time-step to be considered, whereas `u.data[1]` holds the values of `u` for the \"current+1\" time-step.\n",
```

Comment:
- values of the Function.
- held by each node, not cell (typo too)
- `u.data[0]` would put `u.data[0, :,:]` for consistency with coordinates discussion above.
- "given a time-step to be considered" => remove intricated